### PR TITLE
[MM-20400] Migrate 'components/plugin_marketplace/marketplace_item' module and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
In preparation of the upcoming re-design of the Marketplace the whole Marketplace should get a migrated to typescript.

Most of the migration was straight forward. Please see my comments below on the questionable or tricky parts.

Please also note that this PR will go into the Marketplace feature branch.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-20400

#### Related Pull Requests
- Has redux changes https://github.com/mattermost/mattermost-redux/pull/1304

### CODE REVIEW

File: components/plugin_marketplace/marketplace_item/marketplace_item.test.tsx [hunk 1]
```diff
@@ -7,16 +7,15 @@ import {shallow} from 'enzyme';
...
 
 describe('components/MarketplaceItem', () => {
     describe('UpdateDetails', () => {
-        const baseProps = {
+        const baseProps:UpdateDetailsProps = {
             version: '0.0.2',
             releaseNotesUrl: 'http://example.com/release',
             installedVersion: '0.0.1',
             isInstalling: false,
-            isDefaultMarketplace: true,
```
Comment (hanzei) [inline components/plugin_marketplace/marketplace_item/marketplace_item.test.tsx:19]: `UpdateDetails` doesn't have a `isDefaultMarketplace` prop.

File: components/plugin_marketplace/marketplace_item/marketplace_item.test.tsx [hunk 2]
```diff
@@ -36,7 +35,7 @@ describe('components/MarketplaceItem', () => {
             it('when installed version matches available version', () => {
                 const props = {
                     ...baseProps,
-                    installedVersion: baseProps.availableVersion,
+                    installedVersion: baseProps.installedVersion,
```
Comment (hanzei) [inline components/plugin_marketplace/marketplace_item/marketplace_item.test.tsx:38]: I suppose this was a typo. @lieut-data Can you confirm?

File: components/plugin_marketplace/marketplace_item/marketplace_item.test.tsx [hunk 3]
```diff
@@ -195,16 +194,13 @@ describe('components/MarketplaceItem', () => {
...
+        const baseProps:MarketplaceItemProps = {
             id: 'id',
             name: 'name',
             description: 'test plugin',
             version: '1.0.0',
-            downloadUrl: 'http://example.com/download',
-            signatureUrl: 'http://example.com/signature',
             homepageUrl: 'http://example.com',
             installedVersion: '',
-            iconUrl: '',
```
Comment (hanzei) [inline components/plugin_marketplace/marketplace_item/marketplace_item.test.tsx:207]: They are all unused by `MarketplaceItem`, hence I think it's fine to drop them.

File: components/plugin_marketplace/marketplace_item/marketplace_item.test.tsx [hunk 4]
```diff
@@ -36,7 +35,7 @@ describe('components/MarketplaceItem', () => {
             it('when installed version matches available version', () => {
                 const props = {
                     ...baseProps,
-                    installedVersion: baseProps.availableVersion,
+                    installedVersion: baseProps.version,
```
Comment (hanzei) [inline components/plugin_marketplace/marketplace_item/marketplace_item.test.tsx:38]: I suppose this was a typo. @lieut-data Can you confirm?

Comment (lieut-data) [reply components/plugin_marketplace/marketplace_item/marketplace_item.test.tsx:38]: Indeed, looks like it!

**Review by lieut-data — APPROVED** [review]
I'm not an authority on all-things-Typescript, but I see nothing that concerns me from the marketplace side of things. Thanks, Ben!

**Comment by hanzei** [comment]
@mickmister May I request that you take a look at this PR soon as it blocks migrating the rest of the Marketplace code to typescript?

**Review by mickmister — APPROVED** [review]
LGTM :+1: though I have the same non-blocking suggestions I expressed here https://github.com/mattermost/mattermost-webapp/pull/7131

**Comment by mickmister** [comment]
Looks like CI is showing two type check errors

```
components/plugin_marketplace/marketplace_item/index.ts:21:51 - error TS2339: Property 'IsDefaultMarketplace' does not exist on type 'Partial<ClientConfig>'.

21     const isDefaultMarketplace = getConfig(state).IsDefaultMarketplace === 'true';
                                                     ~~~~~~~~~~~~~~~~~~~~

components/plugin_marketplace/marketplace_item/marketplace_item.tsx:13:9 - error TS2305: Module '"../../../node_modules/mattermost-redux/types/plugins"' has no exported member 'MarketplaceLabel'.

13 import {MarketplaceLabel} from 'mattermost-redux/types/plugins';
           ~~~~~~~~~~~~~~~~
```

**Comment by hanzei** [comment]
@mickmister Yes, that CI issue is caused by the missing redux PR: https://github.com/mattermost/mattermost-redux/pull/1304

**Comment by hanzei** [comment]
Skipping QA as this changes goes into a feature branch

**Pull request merged**